timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/AutobatchingPaxosLearnerNetworkClientFactory.java

```diff
@@ -87,7 +89,19 @@ private AutobatchingPaxosLearnerNetworkClient(Client client) {
 
         @Override
         public void learn(long seq, PaxosValue value) {
-
+            Preconditions.checkArgument(seq == value.getRound(), "seq differs from PaxosValue.round");
+            try {
+                learn.apply(Maps.immutableEntry(client, value)).get();
+            } catch (InterruptedException e) {
+                Throwable cause = e.getCause();
+                if (cause instanceof RuntimeException) {
+                    throw (RuntimeException) cause;
+                }
+                throw new RuntimeException(cause);
+            } catch (ExecutionException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            }
```

Comment:
Whoops, this needs to be the other way round (the first block is for `ExecutionException` and the second is for `InterruptedException`)!